app/HutchManager/Controllers/AgentsController.cs

```diff
@@ -79,4 +78,9 @@ public async Task<IActionResult> Delete(int id)
     }
     return NoContent();
   }
+  
+  [HttpPut("{id}/secret")] //api/agents/{id}/secret
+  public async Task<ManageAgent> Set (int id)
```

Comment:
Should this be using Set? or GenerateNewSecret?

Reply:
I think it should be GenerateNewSecret. I will update this now.